### PR TITLE
#1579 Hamburger tweaks

### LINKED ISSUE #1579: Hamburger menu -- remaining tweaks
These are all pretty much cosmetic, and are just suggestions -- might need a little discussion.

This is how it currently stands:

Screenshot 2023-11-01 at 11 51 15 PM

- [x] ~~Doesn't need to be as wide, can just contain the list, that should be enough.~~ As discussed in meeting, make it full width (it's currently doesn't quite go all the way over)
- [x] Be good if it animated (slide) in and out. ~~Maybe we should wrap it in a Semantic [Sidebar](https://react.semantic-ui.com/modules/sidebar/) component?~~
- [x] ~~Should it be on the right-hand side, to align with the Hamburger icon? (I realise this might make the sub-menus over-complicated as they would go off screen by default, so might not be worth the hassle)~~
- [x] Slightly wider spacing between items in list
- [x] The drop-downs that pop up seem a bit "big" in this context. Can we lower the font-size perhaps for the drop-downs/sub-menus? The font-size and spacing of the sub-menus should match the font-size and spacing of the main list. (Done in #1593)
- [x] Should we hide the "Configuration" (and maybe the "Manage") menu completely on Mobile?
- [x]  #1590 PR: #1591 Need to change the background colour to use the same constant as the header bar (as it's configurable in prefs):
Screenshot 2023-11-02 at 4 39 42 PM

### PR DESCRIPTION
**Pull request by CarlosNZ**
Fix #1579 

Minor tweaks to the hamburger menu

### CODE REVIEW

File: semantic/src/site/globals/site.overrides [hunk 1]
```diff
@@ -1835,6 +1835,10 @@ a:hover {
 #data-view-config-panel {
   padding-top: 20px;
   width: 50%;
+  @media only screen and (max-width: @largestMobileScreen) {
+    width: unset;
+    max-width: 100%;
+  }
```
Comment (CarlosNZ) [inline semantic/src/site/globals/site.overrides:1841]: This just makes the Data Views and System Preferences config pages appear slightly better on mobile. Not strictly necessary, but was simple enough.

File: semantic/src/site/globals/site.overrides [hunk 2]
```diff
@@ -2688,14 +2692,20 @@ h6:first-child {
 
   #menu-bar {
     font-size: inherit;
+    position: absolute;
+    top: @mobileHeaderHeight;
+    left: 0px;
+    width: 100%;
+    background-color: inherit;
```
Comment (CarlosNZ) [inline semantic/src/site/globals/site.overrides:2699]: The `inherit` makes the list take the same color as the main header bar (doesn't work properly here, but does in #1591

File: semantic/src/site/globals/site.overrides [hunk 3]
```diff
@@ -2723,6 +2733,7 @@ h6:first-child {
     height: 3px;
     background: @userAreaText;
     margin: 0 0 3px;
+    transition-duration: 0.1s;
```
Comment (CarlosNZ) [inline semantic/src/site/globals/site.overrides:2736]: Hamburger X animates out too

File: src/components/Admin/AdminPreferences.tsx [hunk 4]
```diff
@@ -66,6 +68,7 @@ export const AdminPreferences: React.FC = () => {
           maxWidth={650}
           restrictDelete={({ level }) => level === 1}
           restrictAdd={({ level }) => level === 0}
+          indent={isMobile ? 2 : 4}
```
Comment (CarlosNZ) [inline src/components/Admin/AdminPreferences.tsx:71]: Again, not really necessary, but simple enough.

**Review by guatemartin — APPROVED** [review]
Looks good.

**Pull request merged**